namerd/storage/in-memory/src/main/scala/io/buoyant/namerd/storage/InMemoryDtabStore.scala

```diff
@@ -47,6 +47,19 @@ class InMemoryDtabStore(namespaces: Map[String, Dtab]) extends DtabStore {
     }
   }
 
+  def delete(ns: String): Future[Unit] = {
+    val state = get(ns)
+    dtabStatesMu.synchronized {
+      state.sample match {
+        case Some(_) =>
+          state.update(None)
+          Future.Unit
+        case None =>
+          Future.exception(new DtabNamespaceDoesNotExistException(ns))
+      }
+    }
+  }
```

Comment:
I don't think we want to remove this from the cache.  Anyone observing the discarded state var will no longer receive updates (eg if the namespace is recreated).  Instead, I fixed `list` to only list non-empty namespaces.